bot/seasons/evergreen/bookmark.py

```diff
@@ -57,6 +61,28 @@ def __init__(self, bot: commands.Bot):
         else:
             log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'")
             await ctx.message.add_reaction(Emojis.envelope)
+            await ctx.message.add_reaction(Emojis.pin)
+
+        embed.add_field(
+            name=f'Bookmarked from {ctx.author.name}.',
+            value=f'[Visit original message]({ctx.message.jump_url})',
+            inline=False
+        )
+
+        def check(reaction: discord.Reaction, user: discord.User) -> bool:
+            return reaction.emoji == Emojis.pin
+
+        sent_person = [ctx.author]  # list of id who got the message
+        while True:
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", timeout=60.0, check=check)
+                if user != self.bot.user and user not in sent_person:
+                    log.info(f"{user.name} bookmarked {target_message.jump_url} with title '{title}' from {ctx.author}")
+                    await user.send(embed=embed)
+                    sent_person.append(user)
+            except asyncio.TimeoutError:
+                await ctx.message.clear_reactions()
+                return

```

Comment:
It may be a rare case, but in case something else happen ( user is a bot spamming the reaction / bot cannot send DM to user ) you should include a broad except here, or better, do specific except for when the bot failed to send the DM, since it will effectively crash the entire loop.

Reply:
Broad exceptions should be avoided. If there's another exception type to take into consideration, then it's better to add that specific one in.

In this case, the scope is a bit incorrect. You'd want to ignore those who have DMs disabled, rather than end the loop entirely and clear reactions. So the sending of the DM should be wrapped in it's own try except block catching a `discord.Forbidden`.

Good thought though to consider it; I didn't think of that case.

Reply:
In fact, if the DM sent raises and it just ends up being `pass`ed then instead it's better to use [`contextlib.suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress):
```py
import contextlib
...
with contextlib.suppress(discord.Forbidden):
    await user.send(embed=embed)
sent_person.append(user)
```

Reply:
Why was this conversation resolved? I don't see any changes to code.